eth-rpc/client.nim

```diff
@@ -1,43 +1,46 @@
-import asyncnet, asyncdispatch, tables, json, macros
+import tables, json, macros
+import asyncdispatch2
 import jsonmarshal
 
 type
   RpcClient* = ref object
-    socket: AsyncSocket
+    transp: StreamTransport
     awaiting: Table[string, Future[Response]]
-    address: string
-    port: Port
+    address: TransportAddress
     nextId: int64
   Response* = tuple[error: bool, result: JsonNode]
 
-
 proc newRpcClient*(): RpcClient =
-  ## Creates a new ``RpcClient`` instance. 
-  RpcClient(
-    socket: newAsyncSocket(),
-    awaiting: initTable[string, Future[Response]](),
-    nextId: 1
-  )
+  ## Creates a new ``RpcClient`` instance.
+  result = RpcClient(awaiting: initTable[string, Future[Response]](), nextId: 1)
 
-proc call*(self: RpcClient, name: string, params: JsonNode): Future[Response] {.async.} =
+proc call*(self: RpcClient, name: string,
+           params: JsonNode): Future[Response] {.async.} =
   ## Remotely calls the specified RPC method.
   let id = $self.nextId
   self.nextId.inc
-  let msg = $ %{"jsonrpc": %"2.0", "method": %name, "params": params, "id": %id} & "\c\l"
-  await self.socket.send(msg)
+  let msg = $ %{"jsonrpc": %"2.0", "method": %name, "params": params,
+                "id": %id} & "\c\l"
+  let res = await self.transp.write(msg)
+  assert(res == len(msg))
 
   # completed by processMessage.
   var newFut = newFuture[Response]()
   # add to awaiting responses
   self.awaiting[id] = newFut
   result = await newFut
 
-macro checkGet(node: JsonNode, fieldName: string, jKind: static[JsonNodeKind]): untyped =
+macro checkGet(node: JsonNode, fieldName: string,
+               jKind: static[JsonNodeKind]): untyped =
   let n = genSym(ident = "n") #`node`{`fieldName`}
   result = quote:
     let `n` = `node`{`fieldname`}
-    if `n`.isNil: raise newException(ValueError, "Message is missing required field \"" & `fieldName` & "\"")
-    if `n`.kind != `jKind`.JsonNodeKind: raise newException(ValueError, "Expected " & $(`jKind`.JsonNodeKind) & ", got " & $`node`[`fieldName`].kind)
+    if `n`.isNil:
+      raise newException(ValueError,
+                    "Message is missing required field \"" & `fieldName` & "\"")
```

Comment:
There was an attempt to improve the formatting here, but this new formatting looks quite unusual and strange. Is this the reason why Yuriy is asking about the NEP-1 80-char line limit? :P